File: src/openfl/display/InteractiveObject.nim
import ./DisplayObject

type
  InteractiveObject* = ref object of DisplayObject
    doubleClickEnabled*: bool
    mouseEnabled*: bool
    tabEnabledInternal: bool

proc initInteractiveObject*(self: InteractiveObject) =
  self.initDisplayObject()
  self.doubleClickEnabled = false
  self.mouseEnabled = true
  self.tabEnabledInternal = false

proc newInteractiveObject*(): InteractiveObject =
  let self = InteractiveObject()
  self.initInteractiveObject()
  return self

File: kfl/src/main/kotlin/openfl/display/InteractiveObject.kt
package openfl.display

import openfl.events.Event
import openfl.geom.Rectangle

abstract class InteractiveObject : DisplayObject() {

    var doubleClickEnabled: Boolean = false
    var mouseEnabled: Boolean = true
    var tabEnabled: Boolean = false
        set(value) {
            if (field != value) {
                field = value
                dispatchEvent(Event(Event.TAB_ENABLED_CHANGE, true, false))
            }
        }

    var tabIndex: Int = -1
        set(value) {
            if (field != value) {
                field = value
                dispatchEvent(Event(Event.TAB_INDEX_CHANGE, true, false))
            }
        }

    init {
        __drawableType = openfl.display._internal.IBitmapDrawableType.DISPLAY_OBJECT
    }
}
